7.External-Audio-Device/External-Audio-Device/OTDefaultAudioDevice.m

```diff
@@ -1094,6 +1085,20 @@ - (BOOL)configureAudioSessionWithDesiredAudioRoute:(NSString *)desiredAudioRoute
                                         error:&err];
     }
     
+    if([AUDIO_DEVICE_HEADSET isEqualToString: desiredAudioRoute]) {
```

Comment:
This seems like the only actual change other than the reordering of the `setActive:` calls. Re-add the removed iOS7 guards around `setActive:`, you can keep the reordering, otherwise lgtm.